programs/mpl-core/src/plugins/utils.rs

```diff
@@ -266,6 +269,82 @@ pub fn initialize_plugin<'a, T: DataBlob + SolanaAccount>(
     Ok(())
 }
 
+/// Add an external plugin to the registry and initialize it.
+pub fn initialize_external_plugin<'a, T: DataBlob + SolanaAccount>(
+    init_info: &ExternalPluginInitInfo,
+    plugin_header: &mut PluginHeaderV1,
+    plugin_registry: &mut PluginRegistryV1,
+    account: &AccountInfo<'a>,
+    payer: &AccountInfo<'a>,
+    system_program: &AccountInfo<'a>,
+) -> ProgramResult {
+    let core = T::load(account, 0)?;
+    let header_offset = core.get_size();
+    let plugin_type = init_info.into();
+
+    let (authority, lifecycle_checks) = match &init_info {
+        ExternalPluginInitInfo::LifecycleHook(init_info) => {
+            (init_info.init_plugin_authority, &init_info.lifecycle_checks)
+        }
+        ExternalPluginInitInfo::Oracle(init_info) => {
+            (init_info.init_plugin_authority, &init_info.lifecycle_checks)
+        }
+        ExternalPluginInitInfo::DataStore(init_info) => (init_info.init_plugin_authority, &None),
+    };
+
+    let old_registry_offset = plugin_header.plugin_registry_offset;
+
+    let new_registry_record = ExternalRegistryRecord {
+        plugin_type,
+        authority: authority.unwrap_or(Authority::UpdateAuthority),
+        lifecycle_checks: lifecycle_checks.clone(),
+        offset: old_registry_offset,
+    };
+
+    let mut plugin = ExternalPlugin::from(init_info);
+
+    // If the plugin is a LifecycleHook or DataStore, then we need to set the data offset and length.
+    match &mut plugin {
+        ExternalPlugin::LifecycleHook(hook) => {
+            hook.data_offset = old_registry_offset;
+            hook.data_len = 0;
+        }
+        ExternalPlugin::DataStore(data_store) => {
+            data_store.data_offset = old_registry_offset;
+            data_store.data_len = 0;
+        }
+        _ => {}
+    };
+
+    let plugin_data = plugin.try_to_vec()?;
+    let plugin_size = plugin_data.len();
```

Comment:
Terminology gets a bit strange here since this isn't really like the plugin data length, but I guess its fine.  Just have to use brain a bit when reading the code because internally the plugins have another data len.